### DISCUSSION TITLE
Running phoenixd from systemd

### DISCUSSION THREAD

**Discussion by tsmith123** [ACINQ/phoenixd#26] [General]
Just wondered if anyone has had any luck running phoenixd from a systemd file. I first tried on Fedora but because it's SELinux it was causing some issues. Switched to Ubuntu and still can't get systemd to run the file.

Here's my system file:

```
[Unit]
Description=phonenixd
After=network.target

[Service]
ExecStart=/phoenixd --agree-to-terms-of-service
Restart=always

[Install]
WantedBy=multi-user.target
```

For the `ExecStart` part I've tried `/root/phoenixd` too and also running it from `/usr/bin` and `/usr/local/bin`.

The program runs fine from terminal using `./phoenixd` but refuses to work when run from systemd file.

Anyone had any success in this area?

**Comment by pm47** [comment]
Try gathering infos about the failure, e.g. https://unix.stackexchange.com/questions/237749/where-to-find-the-logs-of-the-services-started-by-systemd-at-computer-boot

**Comment by tsmith123** [comment]
Yeah sorry about the lack of detail with the actual errors. Getting logs from systemd is not easy.

Anyway, sorted it by adding a `User=root` to the Service section. Not sure why it fixes it because the default is `root` anyway but that's the way it goes.

**Comment by sat-hoarder** [comment]
I know you already found resolution but here is what I'm using for my systemd service file.  The caveat is that phoenixd initially needed to be started in an interactive session so that it could create it's necessary support files and the whole agree to terms stuff.  I run it as limited user.  *edit-I just added the After/Wants=nework-online.target just now noticing that I had been sloppy initially creating the service

[Unit]
Description=phoneixd
After=network-online.target
Wants=network-online.target

[Service]After=network-online.target
Wants=network-online.target
WorkingDirectory=/home/phoenixd
ExecStart=/usr/local/bin/phoenixd
User=phoenixd
Restart=always
TimeoutSec=120
RestartSec=30

[Install]
WantedBy=multi-user.target

**Reply by pm47** [reply]
>  The caveat is that phoenixd initially needed to be started in an interactive session so that it could create it's necessary support files and the whole agree to terms stuff

You can do this non-interactively since https://github.com/ACINQ/phoenixd/pull/5.
